lib/_stream_writable.js

```diff
@@ -145,6 +145,22 @@ function WritableState(options, stream, isDuplex) {
   // this must be 0 before 'finish' can be emitted
   this.pendingcb = 0;
 
+  // Stream is still being constructed and no operations
+  // can take place until construction finished.
+  // Async construction is opt in, therefore we start as
+  // ready.
+  this.ready = true;
+
+  // Stream is still being constructed and cannot be
+  // destroyed until construction finished or failed.
+  // Async construction is opt in, therefore we start as
+  // constructed.
+  this.constructing = false;
+
+  // Callback to continue destruction after constrution
+  // has finished or failed.
+  this.destroyCallback = null;
```

Comment:
This is only used in edge cases so we might consider not initialising it here to save some memory in exchange for slower edge case.